CDN-in-a-Box Makefile no longer builds RPMs on non-RHEL systems

## This Bug Report affects these Traffic Control components:
- CDN in a Box

## Current behavior:
`make` checks for a build architecture suffix based on the evaluation of the RPM macro `%_arch`, presumably in an effort to support Apple's new CPU architecture (?). This means that on systems without `rpm` installed, it winds up looking for an output file at `../../dist/{{component}}-{{version}}-{{number of commits}}.{{commit hash}}.{{Enterprise Linux Release Version}}..rpm`, because the architecture is left as an empty string.

## Expected behavior:
`make` should make everything necessary to build CDN-in-a-Box

## Steps to reproduce:
1. `make`

I didn't do `--debug`, but the issue doesn't seem unclear enough to warrant it IMO:
```
Results in 'dist':
total 31880
-rw-r--r-- 1 ocket8888 ocket8888    92212 Jan  3 12:17 build-traffic_ops.log
-rw-r--r-- 1 ocket8888 ocket8888 16815360 Jan  3 12:17 traffic_ops-7.1.0-12307.02d4a142.el8.src.rpm
-rw-r--r-- 1 ocket8888 ocket8888 15731400 Jan  3 12:17 traffic_ops-7.1.0-12307.02d4a142.el8.x86_64.rpm
cp -f "../../dist/traffic_ops-7.1.0-12307.02d4a142.el8..rpm" "traffic_ops/traffic_ops.rpm"
cp: cannot stat '../../dist/traffic_ops-7.1.0-12307.02d4a142.el8..rpm': No such file or directory
make: *** [Makefile:153: traffic_ops/traffic_ops.rpm] Error 1
```

--- a/infrastructure/cdn-in-a-box/Makefile
+++ b/infrastructure/cdn-in-a-box/Makefile
@@ -27,6 +27,12 @@ else
 	RHEL_VERSION := $(shell echo $(RHEL_VERSION) | cut -f1 -d.)
 endif
 
+ifeq ($(shell echo $(RHEL_VERSION) | grep -Eo '^[0-9]+'),7)
+	export BASE_IMAGE := centos
+else
+	BASE_IMAGE := rockylinux
+endif
+
 CIAB_DIR_RELATIVE := $(dir $(MAKEFILE_LIST))
 CIAB_DIR_ABSOLUTE := $(shell cd $(CIAB_DIR_RELATIVE) && pwd)
 TC_DIR := $(CIAB_DIR_RELATIVE)../..
@@ -35,7 +41,8 @@ PKG_COMMAND := $(TC_DIR)/pkg
 PKG_FLAGS := -v -$(RHEL_VERSION)
 BUILD_SUFFIX := _build
 BUILD_NUMBER := $(shell git rev-list HEAD 2>/dev/null | wc -l | tr -d '[[:space:]]').$(shell git rev-parse --short=8 HEAD)
-BUILD_ARCH   := $(shell rpm --eval %_arch)
+# no --rm on the docker command to speed up reuse for shell completion
+BUILD_ARCH   := $(shell docker run --name=ciab-get-$(BASE_IMAGE)-$(RHEL_VERSION)-arch $(BASE_IMAGE):$(RHEL_VERSION) rpm --eval %_arch)
 TC_VERSION := $(shell cat "$(TC_DIR)/VERSION")
 TOMCAT_VERSION := $(shell grep '^\s*TOMCAT_VERSION=' "$(TC_DIR)/traffic_router/build/build_rpm.sh"  | cut -d= -f2)
 TOMCAT_RELEASE := $(shell grep '^\s*TOMCAT_RELEASE=' "$(TC_DIR)/traffic_router/build/build_rpm.sh"  | cut -d= -f2)

